packages/core/src/sdk/product/usePageProductsQuery.ts

```diff
@@ -77,42 +77,32 @@ export const useProductsPrefetch = (page: number | null) => {
 }
 
 function updatesProductsPerPageRef(
-  productsPerPageRef: MutableRefObject<ProductsPerPage[]>,
+  productsPerPage: ProductsPerPage[] = [],
   page: number,
   data: ClientProductsQueryQuery
 ) {
-  const productsPerPage = productsPerPageRef.current
-  const currentProductsPerPage = { page, data }
-
-  if (productsPerPage.length === 0) {
-    productsPerPage.push(currentProductsPerPage)
-    return { productsPerPage, currentProductsPerPage }
-  }
+  const currentPagedProducts = { page, data }
 
   const index = productsPerPage.findIndex((item) => item.page === page)
-  const lastItem = productsPerPage[productsPerPage.length - 1]
-  const shouldReplaceItem = index !== -1
-  const shouldAddNext = page > lastItem.page
+  const shouldReplacePage = index !== -1
 
-  if (shouldReplaceItem) {
-    productsPerPage.splice(index, 1, currentProductsPerPage)
-  } else if (shouldAddNext) {
-    productsPerPage.push(currentProductsPerPage)
+  if (shouldReplacePage) {
+    productsPerPage[index] = currentPagedProducts
   } else {
-    productsPerPage.unshift(currentProductsPerPage)
+    productsPerPage.push(currentPagedProducts)
+    productsPerPage.sort((a, b) => a.page - b.page)
```

Comment:
You're also mutating here

Reply:
We addressed this with pair programming :)